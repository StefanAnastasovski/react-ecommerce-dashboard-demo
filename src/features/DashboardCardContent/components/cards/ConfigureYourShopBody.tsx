import React from "react";

import { Box } from "@mui/material";

import {
  LINK_N_TEXT_FONT_SIZE,
  CARD_COMPLETED_TEXT,
  FONT_WEIGHT_600,
  HEADER_BUTTON_FONT_SIZE,
  WELCOME_MESSAGE_FONT_SIZE,
  FONT_WEIGHT_500,
} from "../../../../data/constants";

import { DASHBOARD_API } from "../../../../data/dashboardConfig";

import Paragraph from "../../../../components/Typography/Paragraph";

const ConfigureYourShopBody = () => {
  const styledCompleted = {
    color: CARD_COMPLETED_TEXT,
    fontSize: WELCOME_MESSAGE_FONT_SIZE,
    fontWeight: FONT_WEIGHT_600,
    marginTop: 3,
  };

  const styledText = {
    color: CARD_COMPLETED_TEXT,
    fontSize: HEADER_BUTTON_FONT_SIZE,
    fontWeight: FONT_WEIGHT_500,
    display: "block",
    marginTop: -1.5,
  };

  const {
    configure_your_shop: { completed },
  } = DASHBOARD_API;

  const completedMessage =
    completed < 100
      ? "Complete all the steps to have a complete shop to best present to your customers."
      : "You complete all the steps.";

  return (
    <>
      <Paragraph styleProps={styledCompleted}>
        {completed}%
        <Box component="span" {...styledText}>
          Completed
        </Box>
      </Paragraph>
      <Paragraph styleProps={{fontSize: LINK_N_TEXT_FONT_SIZE, margin: "1em 0"}}>
        {completedMessage}
      </Paragraph>
    </>
  );
};

export default ConfigureYourShopBody;
