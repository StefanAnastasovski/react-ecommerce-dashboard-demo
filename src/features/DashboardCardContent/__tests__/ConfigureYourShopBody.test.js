import React from "react";
import { shallow } from "enzyme";
import { render } from "@testing-library/react";

import ConfigureYourShopBody from "../components/cards/ConfigureYourShopBody";

describe("<ConfigureYourShopBody />", () => {
  describe("render()", () => {
    it.skip("Shallow rendering - MatchSnapshot", () => {
      const component = shallow(<ConfigureYourShopBody />);
      expect(component).toMatchSnapshot();
    });
    test.skip("Configure your shop card render testing", () => {
      const { getByText } = render(<ConfigureYourShopBody />);
      expect(
        getByText("complete all the steps", { exact: false })
        // "Complete all the steps to have a complete shop to best present to your customers."
      ).toBeDefined();
      expect(getByText("Completed", { exact: true })).toBeDefined();
    });
  });
});
