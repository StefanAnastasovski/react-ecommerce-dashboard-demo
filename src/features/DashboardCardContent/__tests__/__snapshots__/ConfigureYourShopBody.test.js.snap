// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ConfigureYourShopBody /> render() Shallow rendering - MatchSnapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ConfigureYourShopBody />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Paragraph
          styleProps={
            Object {
              "color": "#FFA26B",
              "fontSize": "34px",
              "fontWeight": "600",
              "marginTop": 3,
            }
          }
        >
          45
          %
          <ForwardRef(Box)
            color="#FFA26B"
            component="span"
            display="block"
            fontSize="15px"
            fontWeight="500"
            marginTop={-1.5}
          >
            Completed
          </ForwardRef(Box)>
        </Paragraph>,
        <Paragraph
          styleProps={
            Object {
              "fontSize": "17px",
              "margin": "1em 0",
            }
          }
        >
          Complete all the steps to have a complete shop to best present to your customers.
        </Paragraph>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            45,
            "%",
            <ForwardRef(Box)
              color="#FFA26B"
              component="span"
              display="block"
              fontSize="15px"
              fontWeight="500"
              marginTop={-1.5}
            >
              Completed
            </ForwardRef(Box)>,
          ],
          "styleProps": Object {
            "color": "#FFA26B",
            "fontSize": "34px",
            "fontWeight": "600",
            "marginTop": 3,
          },
        },
        "ref": null,
        "rendered": Array [
          45,
          "%",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Completed",
              "color": "#FFA26B",
              "component": "span",
              "display": "block",
              "fontSize": "15px",
              "fontWeight": "500",
              "marginTop": -1.5,
            },
            "ref": null,
            "rendered": "Completed",
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "propTypes": Object {
                "children": [Function],
                "component": [Function],
                "sx": [Function],
              },
              "render": [Function],
            },
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Complete all the steps to have a complete shop to best present to your customers.",
          "styleProps": Object {
            "fontSize": "17px",
            "margin": "1em 0",
          },
        },
        "ref": null,
        "rendered": "Complete all the steps to have a complete shop to best present to your customers.",
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Paragraph
            styleProps={
              Object {
                "color": "#FFA26B",
                "fontSize": "34px",
                "fontWeight": "600",
                "marginTop": 3,
              }
            }
          >
            45
            %
            <ForwardRef(Box)
              color="#FFA26B"
              component="span"
              display="block"
              fontSize="15px"
              fontWeight="500"
              marginTop={-1.5}
            >
              Completed
            </ForwardRef(Box)>
          </Paragraph>,
          <Paragraph
            styleProps={
              Object {
                "fontSize": "17px",
                "margin": "1em 0",
              }
            }
          >
            Complete all the steps to have a complete shop to best present to your customers.
          </Paragraph>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              45,
              "%",
              <ForwardRef(Box)
                color="#FFA26B"
                component="span"
                display="block"
                fontSize="15px"
                fontWeight="500"
                marginTop={-1.5}
              >
                Completed
              </ForwardRef(Box)>,
            ],
            "styleProps": Object {
              "color": "#FFA26B",
              "fontSize": "34px",
              "fontWeight": "600",
              "marginTop": 3,
            },
          },
          "ref": null,
          "rendered": Array [
            45,
            "%",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Completed",
                "color": "#FFA26B",
                "component": "span",
                "display": "block",
                "fontSize": "15px",
                "fontWeight": "500",
                "marginTop": -1.5,
              },
              "ref": null,
              "rendered": "Completed",
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "propTypes": Object {
                  "children": [Function],
                  "component": [Function],
                  "sx": [Function],
                },
                "render": [Function],
              },
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Complete all the steps to have a complete shop to best present to your customers.",
            "styleProps": Object {
              "fontSize": "17px",
              "margin": "1em 0",
            },
          },
          "ref": null,
          "rendered": "Complete all the steps to have a complete shop to best present to your customers.",
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
